File: pkg/api/apigen/repository.go
package apigen

import "strings"

func SplitRepositoryID(repository string) (string, string) {
	owner, name, _ := strings.Cut(repository, "/")
	return owner, name
}

func RepositoryOwner(repository string) string {
	owner, _, _ := strings.Cut(repository, "/")
	return owner
}

func RepositoryName(repository string) string {
	_, name, _ := strings.Cut(repository, "/")
	return name
}

File: clients/python/surogate_hub_sdk/repository.py
def split_repository_id(repository: str) -> tuple[str, str]:
    """Split an ``owner/repository`` id for generated API methods."""
    owner, sep, name = repository.partition("/")
    if not sep or not owner or not name:
        raise ValueError("repository must be in '<owner>/<repository>' form")
    return owner, name
